HLTrigger/Timer/interface/ProcessCallGraph.h

```diff
@@ -0,0 +1,193 @@
+#ifndef HLTrigger_Timer_interface_ProcessCallGraph_h
+#define HLTrigger_Timer_interface_ProcessCallGraph_h
+
+/*
+ *
+ */
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <type_traits>
+
+// boost optional (used by boost graph) results in some false positives with -Wmaybe-uninitialized
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmaybe-uninitialized"
+#include <boost/graph/adjacency_list.hpp>
+#include <boost/graph/lookup_edge.hpp>
+#include <boost/graph/subgraph.hpp>
+#pragma GCC diagnostic pop
+
+#include "DataFormats/Provenance/interface/ModuleDescription.h"
+#include "FWCore/ServiceRegistry/interface/PathsAndConsumesOfModulesBase.h"
+#include "FWCore/ServiceRegistry/interface/ProcessContext.h"
+#include "HLTrigger/Timer/interface/EDMModuleType.h"
+
+class ProcessCallGraph {
+public:
+
+  struct node_type {
```

Comment:
by the way, would you consider adding similar `type` and `schedued` fields directly to the `edm::ModuleDescription` class ?